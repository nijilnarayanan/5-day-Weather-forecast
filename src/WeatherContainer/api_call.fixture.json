{
	"cod": "200",
	"message": 0,
	"cnt": 40,
	"list": [
		{
			"dt": 1604340000,
			"main": {
				"temp": 293.93,
				"feels_like": 296.28,
				"temp_min": 293.35,
				"temp_max": 293.93,
				"pressure": 1016,
				"sea_level": 1016,
				"grnd_level": 953,
				"humidity": 83,
				"temp_kf": 0.58
			},
			"weather": [{ "id": 802, "main": "Clouds", "description": "scattered clouds", "icon": "03n" }],
			"clouds": { "all": 39 },
			"wind": { "speed": 0.5, "deg": 28 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-02 18:00:00"
		},
		{
			"dt": 1604350800,
			"main": {
				"temp": 292.89,
				"feels_like": 295.21,
				"temp_min": 292.45,
				"temp_max": 292.89,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 952,
				"humidity": 86,
				"temp_kf": 0.44
			},
			"weather": [{ "id": 801, "main": "Clouds", "description": "few clouds", "icon": "02n" }],
			"clouds": { "all": 11 },
			"wind": { "speed": 0.28, "deg": 43 },
			"visibility": 10000,
			"pop": 0.03,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-02 21:00:00"
		},
		{
			"dt": 1604361600,
			"main": {
				"temp": 293.16,
				"feels_like": 295.47,
				"temp_min": 293.07,
				"temp_max": 293.16,
				"pressure": 1016,
				"sea_level": 1016,
				"grnd_level": 953,
				"humidity": 84,
				"temp_kf": 0.09
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01d" }],
			"clouds": { "all": 7 },
			"wind": { "speed": 0.23, "deg": 37 },
			"visibility": 10000,
			"pop": 0.01,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-03 00:00:00"
		},
		{
			"dt": 1604372400,
			"main": {
				"temp": 299.34,
				"feels_like": 301.53,
				"temp_min": 299.34,
				"temp_max": 299.38,
				"pressure": 1017,
				"sea_level": 1017,
				"grnd_level": 956,
				"humidity": 60,
				"temp_kf": -0.04
			},
			"weather": [{ "id": 803, "main": "Clouds", "description": "broken clouds", "icon": "04d" }],
			"clouds": { "all": 57 },
			"wind": { "speed": 0.74, "deg": 169 },
			"visibility": 10000,
			"pop": 0.04,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-03 03:00:00"
		},
		{
			"dt": 1604383200,
			"main": {
				"temp": 301.57,
				"feels_like": 304.04,
				"temp_min": 301.57,
				"temp_max": 301.57,
				"pressure": 1013,
				"sea_level": 1013,
				"grnd_level": 952,
				"humidity": 58,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 65 },
			"wind": { "speed": 1.31, "deg": 186 },
			"visibility": 10000,
			"pop": 0.21,
			"rain": { "3h": 0.27 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-03 06:00:00"
		},
		{
			"dt": 1604394000,
			"main": {
				"temp": 299.83,
				"feels_like": 303.55,
				"temp_min": 299.83,
				"temp_max": 299.83,
				"pressure": 1011,
				"sea_level": 1011,
				"grnd_level": 950,
				"humidity": 73,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 92 },
			"wind": { "speed": 0.97, "deg": 129 },
			"visibility": 10000,
			"pop": 0.7,
			"rain": { "3h": 1.67 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-03 09:00:00"
		},
		{
			"dt": 1604404800,
			"main": {
				"temp": 294.69,
				"feels_like": 297.74,
				"temp_min": 294.69,
				"temp_max": 294.69,
				"pressure": 1013,
				"sea_level": 1013,
				"grnd_level": 951,
				"humidity": 88,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 66 },
			"wind": { "speed": 0.56, "deg": 70 },
			"visibility": 10000,
			"pop": 0.96,
			"rain": { "3h": 2.18 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-03 12:00:00"
		},
		{
			"dt": 1604415600,
			"main": {
				"temp": 293.89,
				"feels_like": 297.02,
				"temp_min": 293.89,
				"temp_max": 293.89,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 953,
				"humidity": 89,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 36 },
			"wind": { "speed": 0.06, "deg": 233 },
			"visibility": 10000,
			"pop": 0.68,
			"rain": { "3h": 2.31 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-03 15:00:00"
		},
		{
			"dt": 1604426400,
			"main": {
				"temp": 293.5,
				"feels_like": 296.3,
				"temp_min": 293.5,
				"temp_max": 293.5,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 952,
				"humidity": 89,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 65 },
			"wind": { "speed": 0.28, "deg": 106 },
			"visibility": 10000,
			"pop": 0.82,
			"rain": { "3h": 1.2 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-03 18:00:00"
		},
		{
			"dt": 1604437200,
			"main": {
				"temp": 293.49,
				"feels_like": 296.43,
				"temp_min": 293.49,
				"temp_max": 293.49,
				"pressure": 1013,
				"sea_level": 1013,
				"grnd_level": 951,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 100 },
			"wind": { "speed": 0.19, "deg": 183 },
			"visibility": 10000,
			"pop": 0.51,
			"rain": { "3h": 0.21 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-03 21:00:00"
		},
		{
			"dt": 1604448000,
			"main": {
				"temp": 293.84,
				"feels_like": 296.63,
				"temp_min": 293.84,
				"temp_max": 293.84,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 952,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 100 },
			"wind": { "speed": 0.62, "deg": 127 },
			"visibility": 2635,
			"pop": 0.82,
			"rain": { "3h": 1.7 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-04 00:00:00"
		},
		{
			"dt": 1604458800,
			"main": {
				"temp": 295.13,
				"feels_like": 298.34,
				"temp_min": 295.13,
				"temp_max": 295.13,
				"pressure": 1016,
				"sea_level": 1016,
				"grnd_level": 954,
				"humidity": 88,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 99 },
			"wind": { "speed": 0.62, "deg": 122 },
			"visibility": 8169,
			"pop": 0.91,
			"rain": { "3h": 0.75 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-04 03:00:00"
		},
		{
			"dt": 1604469600,
			"main": {
				"temp": 300.23,
				"feels_like": 303.05,
				"temp_min": 300.23,
				"temp_max": 300.23,
				"pressure": 1013,
				"sea_level": 1013,
				"grnd_level": 952,
				"humidity": 65,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 89 },
			"wind": { "speed": 1.2, "deg": 152 },
			"visibility": 10000,
			"pop": 0.93,
			"rain": { "3h": 0.96 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-04 06:00:00"
		},
		{
			"dt": 1604480400,
			"main": {
				"temp": 297.76,
				"feels_like": 302.05,
				"temp_min": 297.76,
				"temp_max": 297.76,
				"pressure": 1010,
				"sea_level": 1010,
				"grnd_level": 949,
				"humidity": 88,
				"temp_kf": 0
			},
			"weather": [{ "id": 501, "main": "Rain", "description": "moderate rain", "icon": "10d" }],
			"clouds": { "all": 41 },
			"wind": { "speed": 0.96, "deg": 125 },
			"visibility": 7765,
			"pop": 0.84,
			"rain": { "3h": 4.03 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-04 09:00:00"
		},
		{
			"dt": 1604491200,
			"main": {
				"temp": 294.18,
				"feels_like": 296.97,
				"temp_min": 294.18,
				"temp_max": 294.18,
				"pressure": 1013,
				"sea_level": 1013,
				"grnd_level": 950,
				"humidity": 89,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 28 },
			"wind": { "speed": 0.72, "deg": 101 },
			"visibility": 10000,
			"pop": 0.84,
			"rain": { "3h": 1.29 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-04 12:00:00"
		},
		{
			"dt": 1604502000,
			"main": {
				"temp": 293.47,
				"feels_like": 296.06,
				"temp_min": 293.47,
				"temp_max": 293.47,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 952,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 803, "main": "Clouds", "description": "broken clouds", "icon": "04n" }],
			"clouds": { "all": 64 },
			"wind": { "speed": 0.68, "deg": 101 },
			"visibility": 10000,
			"pop": 0.37,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-04 15:00:00"
		},
		{
			"dt": 1604512800,
			"main": {
				"temp": 293.13,
				"feels_like": 295.82,
				"temp_min": 293.13,
				"temp_max": 293.13,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 951,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 77 },
			"wind": { "speed": 0.33, "deg": 106 },
			"visibility": 10000,
			"pop": 0.83,
			"rain": { "3h": 1.42 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-04 18:00:00"
		},
		{
			"dt": 1604523600,
			"main": {
				"temp": 293.12,
				"feels_like": 295.66,
				"temp_min": 293.12,
				"temp_max": 293.12,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 951,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10n" }],
			"clouds": { "all": 99 },
			"wind": { "speed": 0.54, "deg": 113 },
			"visibility": 3863,
			"pop": 0.88,
			"rain": { "3h": 1.76 },
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-04 21:00:00"
		},
		{
			"dt": 1604534400,
			"main": {
				"temp": 293.26,
				"feels_like": 295.78,
				"temp_min": 293.26,
				"temp_max": 293.26,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 952,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 99 },
			"wind": { "speed": 0.65, "deg": 117 },
			"visibility": 3840,
			"pop": 0.93,
			"rain": { "3h": 1.26 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-05 00:00:00"
		},
		{
			"dt": 1604545200,
			"main": {
				"temp": 293.91,
				"feels_like": 296.63,
				"temp_min": 293.91,
				"temp_max": 293.91,
				"pressure": 1017,
				"sea_level": 1017,
				"grnd_level": 954,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 100 },
			"wind": { "speed": 0.77, "deg": 118 },
			"visibility": 10000,
			"pop": 0.5,
			"rain": { "3h": 0.32 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-05 03:00:00"
		},
		{
			"dt": 1604556000,
			"main": {
				"temp": 297.25,
				"feels_like": 299.69,
				"temp_min": 297.25,
				"temp_max": 297.25,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 952,
				"humidity": 73,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 97 },
			"wind": { "speed": 1.1, "deg": 130 },
			"visibility": 10000,
			"pop": 0.39,
			"rain": { "3h": 0.37 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-05 06:00:00"
		},
		{
			"dt": 1604566800,
			"main": {
				"temp": 298.26,
				"feels_like": 300.75,
				"temp_min": 298.26,
				"temp_max": 298.26,
				"pressure": 1011,
				"sea_level": 1011,
				"grnd_level": 950,
				"humidity": 68,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 82 },
			"wind": { "speed": 0.92, "deg": 148 },
			"visibility": 10000,
			"pop": 0.41,
			"rain": { "3h": 0.4 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-05 09:00:00"
		},
		{
			"dt": 1604577600,
			"main": {
				"temp": 294.01,
				"feels_like": 296.83,
				"temp_min": 294.01,
				"temp_max": 294.01,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 951,
				"humidity": 86,
				"temp_kf": 0
			},
			"weather": [{ "id": 803, "main": "Clouds", "description": "broken clouds", "icon": "04n" }],
			"clouds": { "all": 71 },
			"wind": { "speed": 0.23, "deg": 33 },
			"visibility": 10000,
			"pop": 0.41,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-05 12:00:00"
		},
		{
			"dt": 1604588400,
			"main": {
				"temp": 292.87,
				"feels_like": 295.36,
				"temp_min": 292.87,
				"temp_max": 292.87,
				"pressure": 1016,
				"sea_level": 1016,
				"grnd_level": 953,
				"humidity": 89,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01n" }],
			"clouds": { "all": 9 },
			"wind": { "speed": 0.34, "deg": 40 },
			"visibility": 10000,
			"pop": 0.11,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-05 15:00:00"
		},
		{
			"dt": 1604599200,
			"main": {
				"temp": 292.23,
				"feels_like": 294.44,
				"temp_min": 292.23,
				"temp_max": 292.23,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 953,
				"humidity": 90,
				"temp_kf": 0
			},
			"weather": [{ "id": 801, "main": "Clouds", "description": "few clouds", "icon": "02n" }],
			"clouds": { "all": 22 },
			"wind": { "speed": 0.47, "deg": 45 },
			"visibility": 10000,
			"pop": 0.01,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-05 18:00:00"
		},
		{
			"dt": 1604610000,
			"main": {
				"temp": 292.9,
				"feels_like": 295.13,
				"temp_min": 292.9,
				"temp_max": 292.9,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 952,
				"humidity": 87,
				"temp_kf": 0
			},
			"weather": [{ "id": 803, "main": "Clouds", "description": "broken clouds", "icon": "04n" }],
			"clouds": { "all": 82 },
			"wind": { "speed": 0.52, "deg": 68 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-05 21:00:00"
		},
		{
			"dt": 1604620800,
			"main": {
				"temp": 293.48,
				"feels_like": 295.78,
				"temp_min": 293.48,
				"temp_max": 293.48,
				"pressure": 1017,
				"sea_level": 1017,
				"grnd_level": 954,
				"humidity": 84,
				"temp_kf": 0
			},
			"weather": [{ "id": 804, "main": "Clouds", "description": "overcast clouds", "icon": "04d" }],
			"clouds": { "all": 91 },
			"wind": { "speed": 0.42, "deg": 81 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-06 00:00:00"
		},
		{
			"dt": 1604631600,
			"main": {
				"temp": 296.78,
				"feels_like": 298.99,
				"temp_min": 296.78,
				"temp_max": 296.78,
				"pressure": 1018,
				"sea_level": 1018,
				"grnd_level": 956,
				"humidity": 70,
				"temp_kf": 0
			},
			"weather": [{ "id": 804, "main": "Clouds", "description": "overcast clouds", "icon": "04d" }],
			"clouds": { "all": 99 },
			"wind": { "speed": 0.73, "deg": 87 },
			"visibility": 10000,
			"pop": 0.16,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-06 03:00:00"
		},
		{
			"dt": 1604642400,
			"main": {
				"temp": 300.48,
				"feels_like": 302.25,
				"temp_min": 300.48,
				"temp_max": 300.48,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 953,
				"humidity": 56,
				"temp_kf": 0
			},
			"weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }],
			"clouds": { "all": 91 },
			"wind": { "speed": 1.33, "deg": 78 },
			"visibility": 10000,
			"pop": 0.3,
			"rain": { "3h": 0.21 },
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-06 06:00:00"
		},
		{
			"dt": 1604653200,
			"main": {
				"temp": 300.88,
				"feels_like": 302.58,
				"temp_min": 300.88,
				"temp_max": 300.88,
				"pressure": 1012,
				"sea_level": 1012,
				"grnd_level": 951,
				"humidity": 53,
				"temp_kf": 0
			},
			"weather": [{ "id": 802, "main": "Clouds", "description": "scattered clouds", "icon": "03d" }],
			"clouds": { "all": 30 },
			"wind": { "speed": 1.12, "deg": 75 },
			"visibility": 10000,
			"pop": 0.31,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-06 09:00:00"
		},
		{
			"dt": 1604664000,
			"main": {
				"temp": 294.52,
				"feels_like": 296.55,
				"temp_min": 294.52,
				"temp_max": 294.52,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 953,
				"humidity": 77,
				"temp_kf": 0
			},
			"weather": [{ "id": 801, "main": "Clouds", "description": "few clouds", "icon": "02n" }],
			"clouds": { "all": 20 },
			"wind": { "speed": 0.6, "deg": 32 },
			"visibility": 10000,
			"pop": 0.22,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-06 12:00:00"
		},
		{
			"dt": 1604674800,
			"main": {
				"temp": 293.04,
				"feels_like": 294.96,
				"temp_min": 293.04,
				"temp_max": 293.04,
				"pressure": 1018,
				"sea_level": 1018,
				"grnd_level": 955,
				"humidity": 83,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01n" }],
			"clouds": { "all": 0 },
			"wind": { "speed": 0.61, "deg": 37 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-06 15:00:00"
		},
		{
			"dt": 1604685600,
			"main": {
				"temp": 292.09,
				"feels_like": 293.77,
				"temp_min": 292.09,
				"temp_max": 292.09,
				"pressure": 1017,
				"sea_level": 1017,
				"grnd_level": 954,
				"humidity": 87,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01n" }],
			"clouds": { "all": 0 },
			"wind": { "speed": 0.84, "deg": 21 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-06 18:00:00"
		},
		{
			"dt": 1604696400,
			"main": {
				"temp": 291.21,
				"feels_like": 292.67,
				"temp_min": 291.21,
				"temp_max": 291.21,
				"pressure": 1017,
				"sea_level": 1017,
				"grnd_level": 954,
				"humidity": 89,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01n" }],
			"clouds": { "all": 5 },
			"wind": { "speed": 0.87, "deg": 24 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-06 21:00:00"
		},
		{
			"dt": 1604707200,
			"main": {
				"temp": 291.48,
				"feels_like": 292.82,
				"temp_min": 291.48,
				"temp_max": 291.48,
				"pressure": 1019,
				"sea_level": 1019,
				"grnd_level": 956,
				"humidity": 86,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01d" }],
			"clouds": { "all": 7 },
			"wind": { "speed": 0.89, "deg": 22 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-07 00:00:00"
		},
		{
			"dt": 1604718000,
			"main": {
				"temp": 299.02,
				"feels_like": 299.52,
				"temp_min": 299.02,
				"temp_max": 299.02,
				"pressure": 1019,
				"sea_level": 1019,
				"grnd_level": 958,
				"humidity": 52,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01d" }],
			"clouds": { "all": 2 },
			"wind": { "speed": 1.73, "deg": 84 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-07 03:00:00"
		},
		{
			"dt": 1604728800,
			"main": {
				"temp": 300.91,
				"feels_like": 301.43,
				"temp_min": 300.91,
				"temp_max": 300.91,
				"pressure": 1015,
				"sea_level": 1015,
				"grnd_level": 954,
				"humidity": 46,
				"temp_kf": 0
			},
			"weather": [{ "id": 801, "main": "Clouds", "description": "few clouds", "icon": "02d" }],
			"clouds": { "all": 24 },
			"wind": { "speed": 1.6, "deg": 86 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-07 06:00:00"
		},
		{
			"dt": 1604739600,
			"main": {
				"temp": 300.4,
				"feels_like": 301.62,
				"temp_min": 300.4,
				"temp_max": 300.4,
				"pressure": 1014,
				"sea_level": 1014,
				"grnd_level": 953,
				"humidity": 51,
				"temp_kf": 0
			},
			"weather": [{ "id": 803, "main": "Clouds", "description": "broken clouds", "icon": "04d" }],
			"clouds": { "all": 62 },
			"wind": { "speed": 1.22, "deg": 77 },
			"visibility": 10000,
			"pop": 0.03,
			"sys": { "pod": "d" },
			"dt_txt": "2020-11-07 09:00:00"
		},
		{
			"dt": 1604750400,
			"main": {
				"temp": 293.72,
				"feels_like": 295.27,
				"temp_min": 293.72,
				"temp_max": 293.72,
				"pressure": 1016,
				"sea_level": 1016,
				"grnd_level": 954,
				"humidity": 78,
				"temp_kf": 0
			},
			"weather": [{ "id": 802, "main": "Clouds", "description": "scattered clouds", "icon": "03n" }],
			"clouds": { "all": 46 },
			"wind": { "speed": 0.95, "deg": 58 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-07 12:00:00"
		},
		{
			"dt": 1604761200,
			"main": {
				"temp": 291.67,
				"feels_like": 293.04,
				"temp_min": 291.67,
				"temp_max": 291.67,
				"pressure": 1018,
				"sea_level": 1018,
				"grnd_level": 955,
				"humidity": 83,
				"temp_kf": 0
			},
			"weather": [{ "id": 800, "main": "Clear", "description": "clear sky", "icon": "01n" }],
			"clouds": { "all": 7 },
			"wind": { "speed": 0.65, "deg": 3 },
			"visibility": 10000,
			"pop": 0,
			"sys": { "pod": "n" },
			"dt_txt": "2020-11-07 15:00:00"
		}
	],
	"city": {
		"id": 1655562,
		"name": "Mai",
		"coord": { "lat": 20.9372, "lon": 101.4119 },
		"country": "LA",
		"population": 3225,
		"timezone": 25200,
		"sunrise": 1604272643,
		"sunset": 1604313501
	}
}
