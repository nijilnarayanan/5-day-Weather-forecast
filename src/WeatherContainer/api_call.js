const API_KEY = 'bab3031dd3ab462fa1ca1dc7107b9eb3'
const CITY = 'mai'
const API_URL = `http://api.openweathermap.org/data/2.5/forecast?q=${CITY}&appid=${API_KEY}`

/* eslint import/no-anonymous-default-export: [2, {"allowArrowFunction": true}] */
export default () => fetch(API_URL).then((res) => res.json())
