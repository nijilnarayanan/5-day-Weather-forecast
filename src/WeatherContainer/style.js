import styled from 'styled-components'

export const Wrapper = styled.section`
	display: flex;
	flex-direction: column;
	margin: 0 16px;

	@media screen and (min-width: 767px) {
		flex-direction: row;
		justify-content: space-between;
		margin: 32px;
	}
`
