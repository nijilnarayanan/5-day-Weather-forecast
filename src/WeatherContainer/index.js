import React, { useEffect, useState } from 'react'
import Card from '../Card'
import { Wrapper } from './style'
import fetchWeatherData from './api_call'

const WeatherContainer = () => {
	const [weather, setWeather] = useState([])

	useEffect(() => {
		fetchWeatherData().then((data) => {
			const fiveDayData = data.list.filter((reading) => reading.dt_txt.includes('18:00:00'))
			setWeather(fiveDayData)
		})
	}, [])

	return (
		<Wrapper>
			{weather && weather.map((weatherData, index) => <Card weatherData={weatherData} key={index} />)}
		</Wrapper>
	)
}

export default WeatherContainer
