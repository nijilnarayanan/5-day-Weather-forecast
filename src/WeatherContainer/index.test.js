import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import WeatherContainer from './index'
import fetchWeatherData from './api_call'

import dataFixture from './api_call.fixture'

jest.mock('./api_call')

test('Run Test <WeatherContainer />', async () => {
	const container = document.createElement('div')
	fetchWeatherData.mockResolvedValue(dataFixture)
	await act(async () => {
		ReactDOM.render(<WeatherContainer />, container)
	})
	expect(pretty(container.innerHTML)).toMatchSnapshot()
})
