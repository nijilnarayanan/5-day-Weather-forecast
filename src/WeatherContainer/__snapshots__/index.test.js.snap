// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run Test <WeatherContainer /> 1`] = `
"<section class=\\"sc-gKsewC jZCmVk\\">
  <article class=\\"sc-bdfBwQ hikiyW\\">
    <div class=\\"sc-gsTCUz jnUaKZ\\">Mon</div>
    <div class=\\"sc-dlfnbm jbeaVV\\">2nd, Nov</div><img src=\\"https://openweathermap.org/img/w/03n.png\\" class=\\"sc-hKgILt\\">
    <div class=\\"sc-eCssSg fajYQx\\">21°C</div>
    <div class=\\"sc-jSgupP cuBoir\\">scattered clouds</div>
  </article>
  <article class=\\"sc-bdfBwQ hikiyW\\">
    <div class=\\"sc-gsTCUz jnUaKZ\\">Tue</div>
    <div class=\\"sc-dlfnbm jbeaVV\\">3rd, Nov</div><img src=\\"https://openweathermap.org/img/w/10n.png\\" class=\\"sc-hKgILt\\">
    <div class=\\"sc-eCssSg fajYQx\\">20°C</div>
    <div class=\\"sc-jSgupP cuBoir\\">light rain</div>
  </article>
  <article class=\\"sc-bdfBwQ hikiyW\\">
    <div class=\\"sc-gsTCUz jnUaKZ\\">Wed</div>
    <div class=\\"sc-dlfnbm jbeaVV\\">4th, Nov</div><img src=\\"https://openweathermap.org/img/w/10n.png\\" class=\\"sc-hKgILt\\">
    <div class=\\"sc-eCssSg fajYQx\\">20°C</div>
    <div class=\\"sc-jSgupP cuBoir\\">light rain</div>
  </article>
  <article class=\\"sc-bdfBwQ hikiyW\\">
    <div class=\\"sc-gsTCUz jnUaKZ\\">Thu</div>
    <div class=\\"sc-dlfnbm jbeaVV\\">5th, Nov</div><img src=\\"https://openweathermap.org/img/w/02n.png\\" class=\\"sc-hKgILt\\">
    <div class=\\"sc-eCssSg fajYQx\\">19°C</div>
    <div class=\\"sc-jSgupP cuBoir\\">few clouds</div>
  </article>
  <article class=\\"sc-bdfBwQ hikiyW\\">
    <div class=\\"sc-gsTCUz jnUaKZ\\">Fri</div>
    <div class=\\"sc-dlfnbm jbeaVV\\">6th, Nov</div><img src=\\"https://openweathermap.org/img/w/01n.png\\" class=\\"sc-hKgILt\\">
    <div class=\\"sc-eCssSg fajYQx\\">19°C</div>
    <div class=\\"sc-jSgupP cuBoir\\">clear sky</div>
  </article>
</section>"
`;
